### PR TITLE
Add new metadata fields to the widgets

### PR DESCRIPTION
**Pull request by clementprdhomme**
This PR adds new metadata fields (Private name, Citation and “Allow download”) to the widget creation and edition forms.

<p align="center">
Edition form now contains inputs and checkboxes for the new metadata fields
</p>

## Testing instructions

Try to create or edit widgets setting different values for the new fields.

## Pivotal Tracker

[Task](https://www.pivotaltracker.com/story/show/170235238).

### CODE REVIEW

File: app/models/site.rb [hunk 1]
```diff
@@ -308,7 +308,8 @@ def generate_slug
   end
 
   def apply_settings
-    system "rake site:apply_settings[#{self.id}] &"
+    #system "rake site:apply_settings[#{self.id}] &"
+    compile_css
```
Comment (santostiago) [inline app/models/site.rb]: @jesusgonzalezrivera why this change for this task?

**Review by santostiago — APPROVED** [review]

**Review by jesusgonzalezrivera — APPROVED** [review]
